content/altinn-studio/guides/monitor-and-instrument/_index.nb.md

```diff
@@ -0,0 +1,121 @@
+
+## Egendefinert instrumentering
+
+Vi illustrerer egendefinert instrumentering med et eksempel. I `Program.cs` legger vi til en simpel `IHostedService` implementasjon
+som kan instrumenteres til å eksponere telemetri.
+
+Telemetri- og instrumentering-API'ene i Altinn.App biblioteket bli eksponert gjennom `Telemetry`-klassen. Det er
+et trådsikkert singleton-objekt tilgjengelig i dependency injection containeren.
+La oss utvide `Program.cs` til å inkludere dette.
+
+Hvis de ikke allerede er der, så trenger vi de følgende avhengighetene på toppen av filen:
+
+```csharp
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Hosting;
+using Altinn.App.Core.Features;
+```
+
+Så kan vi implementere den følgende klassen nederst i filen:
+
+```csharp
+sealed class StartupService(ILogger<StartupService> logger, Telemetry telemetry) : IHostedService
+{
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        // Lag en egendefinert metrikk - en teller i dette tilfellet. Navnet blir `altinn_app_started`
+        var counter = telemetry.Meter.CreateCounter<long>(Telemetry.Metrics.CreateName("started"));
+        // Inkrementer telleren
+        counter.Add(1);
+
+        // Start en aktivitet, som blir utsendt som et OTel span
+        using var activity = telemetry.ActivitySource.StartActivity("StartupService");
+        {
+            // Vent litt, deretter legg til en underaktivitet
+            await Task.Delay(100, cancellationToken);
+            using var childActivity = telemetry.ActivitySource.StartActivity("ChildActivity");
+            await Task.Delay(100, cancellationToken);
+        }
+
+        // Logger kommer fra `ILogger<T>` interfacet
+        var now = DateTimeOffset.UtcNow;
+        logger.LogInformation("StartupService logging - Now={Now}", now);
+    }
+
+    public Task StopAsync(CancellationToken cancellationToken) => Task.CompletedTask;
+}
+```
+
+[Les mer om instrumenteringsmuligheter på referansesiden for instrumentering](/nb/altinn-studio/reference/monitoring/instrumentation).
+
+## Visualisering
+
+Her er en kort oversikt over visualisering av telemetrien vi instrumenterte over.
+
+[Les mer om visualisering av telemetri på referansesiden for visualisering](/nb/altinn-studio/reference/monitoring/visualisation).
+
+### Lokal utvikling
+
+Ved lokal utvikling med [localtest](/nb/altinn-studio/guides/local-dev/), så kan en monitoreringsstack bestående av Grafana og OpenTelemetry Collector
+provisjoneres ved siden av localtest og Platform API'er. [Se localtest README for mer informasjon](https://github.com/Altinn/app-localtest/blob/main/README.md).
+
+Monitoreringsoppsettet i localtest inneholder en Grafana instans med ASP.NET Core dashboard og et preview Altinn app dashboard.
+I tillegg gir det muligheten til å fritt undersøke telemetrien som eksponeres fra Altinn plattform og bibliotek.
```

Comment:
Kjører man lokalt er det vel mock av Altinn platformen sin telemetri som eksponeres? Vet ikke om det er misvisende for sluttbruker om telemetrien man da får i TT skiller seg fra det man har sett lokalt. Mulig det bare må presiseres hva man mener med Altinn plattform.